src/main/java/com/patreon/PatreonOAuth.java

```diff
@@ -0,0 +1,68 @@
+package com.patreon;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import org.jsoup.Jsoup;
+
+import java.io.IOException;
+
+public class PatreonOAuth {
+    private final String clientID;
+    private final String clientSecret;
+    private static final Gson gson = new GsonBuilder().serializeNulls().enableComplexMapKeySerialization().create();
+
+
+    public PatreonOAuth(String clientID, String clientSecret) {
+        this.clientID = clientID;
+        this.clientSecret = clientSecret;
+    }
+
+    public String getDefaultOauthUrl(String redirectUri) {
+        return "https://www.patreon.com/oauth2/authorize?response_type=code&client_id=" + clientID + "&redirect_uri=" + redirectUri;
+    }
+
+    public Token getToken(String code, String redirectURI) throws IOException {
+        return toObject(Jsoup.connect("https://api.patreon.com/oauth2/token")
+                .data("grant_type", "authorization_code")
+                .data("code", code)
+                .data("client_id", clientID)
+                .data("client_secret", clientSecret)
+                .data("redirect_uri", redirectURI).ignoreContentType(true).post().body().text(), Token.class);
+    }
+
+    public Token refreshToken(String refreshToken) throws IOException {
+        return toObject(Jsoup.connect("https://api.patreon.com/oauth2/token")
+                .data("grant_type", "refresh_token")
+                .data("client_id", clientID)
+                .data("client_secret", clientSecret)
+                .data("refresh_token", refreshToken).ignoreContentType(true).post().body().text(), Token.class);
+    }
+
+    public static class Token {
+        private String access_token;
+        private String refresh_token;
+        private int expires_in;
+        private String scope;
+        private String token_type;
+
+        public Token(String access_token, String refresh_token, int expires_in, String scope, String token_type) {
```

Comment:
`gson` supports mapping field names to other capitalization schemes: https://stackoverflow.com/questions/2370745/convert-json-style-properties-names-to-java-camelcase-names-with-gson/11810183#11810183

Reply:
yup, that's what I use for parsing most `PatreonAPI` stuff. how are you recommending we use that here?